lib/lte_link_control/lte_lc.c

```diff
@@ -1384,31 +1445,71 @@ int lte_lc_lte_mode_get(enum lte_lc_lte_mode *mode)
 	return 0;
 }
 
-int lte_lc_neighbor_cell_measurement(enum lte_lc_neighbor_search_type type)
+int lte_lc_neighbor_cell_measurement(struct lte_lc_ncellmeas_params *params)
 {
 	int err;
+	/* lte_lc defaults for the used params */
+	struct lte_lc_ncellmeas_params used_params = {
+		.search_type = LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT,
+		.gci_count = 0,
+	};
+
+	__ASSERT(!IN_RANGE(
+			(int)params,
+			LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT,
+			LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_COMPLETE),
+		 "Invalid argument, API does not accept enum values directly anymore");
+
+	if (ncellmeas_ongoing) {
+		return -EINPROGRESS;
+	}
+
+	if (params != NULL) {
+		used_params = *params;
+	}
 
 	/* Starting from modem firmware v1.3.1, there is an optional parameter to specify
 	 * the type of search.
 	 * If the type is LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT, we therefore use the AT
 	 * command without parameters to avoid error messages for older firmware version.
+	 * Starting from modem firmware v1.3.4, additional CGI search types and
+	 * GCI count are supported.
 	 */
-
-	if (type == LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_LIGHT) {
+	if (used_params.search_type == LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_LIGHT) {
 		err = nrf_modem_at_printf("AT%%NCELLMEAS=1");
-	} else if (type == LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_COMPLETE) {
+	} else if (used_params.search_type == LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_COMPLETE) {
 		err = nrf_modem_at_printf("AT%%NCELLMEAS=2");
+	} else if (used_params.search_type == LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_DEFAULT) {
+		err = nrf_modem_at_printf("AT%%NCELLMEAS=3,%d", used_params.gci_count);
+	} else if (used_params.search_type == LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_LIGHT) {
+		err = nrf_modem_at_printf("AT%%NCELLMEAS=4,%d", used_params.gci_count);
+	} else if (used_params.search_type == LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_EXTENDED_COMPLETE) {
+		err = nrf_modem_at_printf("AT%%NCELLMEAS=5,%d", used_params.gci_count);
 	} else {
 		/* Defaulting to use LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT */
 		err = nrf_modem_at_printf("AT%%NCELLMEAS");
```

Comment:
Maybe explicitly check for `LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT` and then have the final case handle invalid search types with either a debug log or an error return

Reply:
I have left this as it is now.